pkg/hasher/hasher.go

```diff
@@ -8,14 +8,27 @@ import (
 )
 
 // HashPassword returns sha512 hash string for provided password
-func HashPassword(password string) string {
+func HashPassword(password string) (string, error) {
+	var passwordHash string
 	hash := sha512.New()
-	hash.Write([]byte(password))
+	_, err := hash.Write([]byte(password))
 
-	return fmt.Sprintf("%x", hash.Sum(nil))
+	if err != nil {
+		return passwordHash, err
+	}
+
+	passwordHash = fmt.Sprintf("%x", hash.Sum(nil))
```

Comment:
fmt.Sprintf not really necessary - you can do simlply - string(hash.Sum(nil))

Reply:
If I place type casting here tests are failing.
```
say@say-HP-ProBook-440-G6:~/go/src/lgc/pkg/hasher$ go test
--- FAIL: TestHashPassword (0.00s)
    hasher_test.go:8: HashPassword() = "\x9bq\xd2$\xbdb\xf3x]\x96\xd4j\xd3\xea=s1\x9b\xfb\u0089\f\xaa\xda\xe2\xdf\xf7%\x19g<\xa7##\xc3ٛ\xa5\xc1\x1d|z\xccn\x14\xb8\xc5\xda\fFcG\\.\\:\xde\xf4os\xbc\xde\xc0C", want "9b71d224bd62f3785d96d46ad3ea3d73319bfbc2890caadae2dff72519673ca72323c3d99ba5c11d7c7acc6e14b8c5da0c4663475c2e5c3adef46f73bcdec043"
--- FAIL: TestCheckPasswordHash (0.00s)
    hasher_test.go:15: CheckPasswordHash() = false, want true
FAIL
exit status 1
FAIL	github.com/andriystech/lgc/pkg/hasher	0.001s
```